xbmc/addons/AddonManager.cpp

```diff
@@ -448,12 +448,22 @@ VECADDONS CAddonMgr::GetAvailableUpdates()
 
   VECADDONS updates;
   VECADDONS installed;
+  VECADDONS all_addons;
   GetAddons(installed);
+  m_database.GetRepositoryContent(all_addons);
+  std::map<std::string, AddonPtr> last_versions;
+  for (const auto& addon : all_addons)
+  {
+    const auto& latest_known = last_versions.find(addon->ID());
+    if (latest_known == last_versions.end() || addon->Version() > latest_known->second->Version())
+      last_versions[addon->ID()] = std::move(addon);
+  }
+
```

Comment:
`installed` is used first time after the new loop. I would place `GetAddons(installed);` directly before the loop it is first used (like it was before).